### PR TITLE
You should be able to navigate to items that are in generated code.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/8009

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by balajikris** [comment]
I want this to work as well, but simply removing the check will re-activate bug 1174255.

**Comment by CyrusNajmabadi** [comment]
This doesn't appear to be the case.  In d62acfa6d5b66ca9502dd5a2ad42939d1fe8d3f5 @brettfo added this code so SyntaxTreeInfo:

```c#
+                            // We've received a number of error reports where DeclaredSymbolInfo.GetSymbolAsync() will
+                            // crash because the document's syntax root doesn't contain the span of the node returned
+                            // by TryGetDeclaredSymbolInfo().  There are two possibilities for this crash:
+                            //   1) syntaxFacts.TryGetDeclaredSymbolInfo() is returning a bad span, or
+                            //   2) Document.GetSyntaxRootAsync() (called from DeclaredSymbolInfo.GetSymbolAsync) is
+                            //      returning a bad syntax root that doesn't represent the original parsed document.
+                            // By adding the `root.FullSpan.Contains()` check below, if we get similar crash reports in
+                            // the future then we know the problem lies in (2).  If, however, the problem is really in
+                            // TryGetDeclaredSymbolInfo, then this will at least prevent us from returning bad spans
+                            // and will prevent the crash from occuring.
                             DeclaredSymbolInfo declaredSymbolInfo;
-                            if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo))
+                            if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo) &&
+                                root.FullSpan.Contains(declaredSymbolInfo.Span))
                             {
                                 declaredSymbolInfos.Add(declaredSymbolInfo);
                             }
```

**Comment by CyrusNajmabadi** [comment]
I've switched us to using a non-fatal-watson so we can figure out what's going on here.  We should not be breaking features to mask an underlying issue lower in the stack.

**Comment by balajikris** [comment]
I agree, however, i didn't want us to trade a feature with a crash either :) two wrongs don't make it right. I'm happy with the addition of watson, so :+1:

**Comment by brettfo** [comment]
NFW is indeed better than just swallowing the error.  :shipit:

**Comment by jasonmalinowski** [comment]
:+1:. IIRC I wasn't happy with this change originally!

File: src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeInfo.cs [hunk 1]
```diff
@@ -170,10 +171,20 @@ public static Task<SyntaxTreeDeclarationInfo> GetDeclarationInfoAsync(Document d
...
+                                if (root.FullSpan.Contains(declaredSymbolInfo.Span))
+                                {
+                                    declaredSymbolInfos.Add(declaredSymbolInfo);
+                                }
+                                else
+                                {
+                                    var message =
+$@"Invalid span in {nameof(declaredSymbolInfo)}.
+{nameof(declaredSymbolInfo.Span)} = {declaredSymbolInfo.Span}
+{nameof(root.FullSpan)} = {root.FullSpan}";
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeInfo.cs]: Maybe :)

My understanding from before was that the crash was happening much further down hte line.  This pushes the watson much closer to the point that we can identify that something went wrong.  If this is not suffickient, we can hopefully keep pushing things back.

**Pull request merged**